src/ui/public/kfetch/fetch_error.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export class FetchError extends Error {
```

Comment:
I know this is a _tiny but outside of scope_ but can we rename this `KFetchError`? This way it will be really obvious which module it has originated from. I think `FetchError` is just a bit too vague. I mention this because I've introduced a `SearchError` in the [rollups feature branch](https://github.com/elastic/kibana/blob/feature/rollups/src/ui/public/courier/search_strategy/search_error.js) and seeing this error makes me think that the `SearchError` name is too vague too. I'll probably rename it `CourierSearchError` or something.

Doing a rename will require making a change in one other place in the codebase: https://github.com/elastic/kibana/blob/master/src/ui/public/error_auto_create_index/error_auto_create_index.test.js